sound/soc/sof/intel/hda-loader.c

```diff
@@ -369,9 +369,15 @@ int hda_dsp_cl_boot_firmware_iccmax(struct snd_sof_dev *sdev)
 
 static int hda_dsp_boot_imr(struct snd_sof_dev *sdev)
 {
+	const struct sof_intel_dsp_desc *chip_info;
 	int ret;
 
-	ret = cl_dsp_init(sdev, 0, true);
+	chip_info = get_chip_info(sdev->pdata);
+	if (chip_info && chip_info->dsp_init)
```

Comment:
here you test for chip_info being valid but not line 439 below. I don't think we need to test, this is assumed to be set.